lazylru.go

```diff
@@ -74,6 +84,45 @@ func NewT[K comparable, V any](maxItems int, ttl time.Duration) *LazyLRU[K, V] {
 	return lru
 }
 
+// OnEvict registers a callback that will be executed when items are removed
+// from the cache via eviction due to max size or because the TTL has been
+// exceeded. These functions will not be called with a lock and will not block
+// future reaping. Be sure any callback registered can complete the number of
+// expected calls (based on your expire/eviction rates) or you may create a
+// backlog of goroutines.
+//
+// If a Set or MSet operation causes an eviction, this function will be called
+// synchronously to that Set or MSet call.
+func (lru *LazyLRU[K, V]) OnEvict(cb EvictCB[K, V]) {
+	lru.lock.Lock()
+	lru.onEvict = append(lru.onEvict, cb)
+	lru.numEvictCB++
+	lru.lock.Unlock()
+}
+
+func (lru *LazyLRU[K, V]) execOnEvict(deathList []*item[K, V]) {
+	if len(deathList) == 0 {
+		return
+	}
+	if atomic.LoadInt32(&(lru.numEvictCB)) == 0 {
```

Comment:
nit: Any reason to do this over just storing an `atomic.Int32` to begin with? Should be outcome / sizing on the struct, just with better ergonomic/semantics when used.

Reply:
There is a reason, but it's a little silly. When callbacks are registered, we have to take a lock to safely add to the slice. This value is just a reflection of the length of that slice, so it is incremented at the same time. Since I'm just incrementing the value, I don't use the atomic operators to do it. 

As a practical matter, I shouldn't care because the number of times that the `OnEvict` registration function is called should be very small and should likely occur before any items are added (let alone removed) from the LRU.

Given that there is more chance that I or some future developer forgets to read atomically from the value if it is an `int32` rather than `atomic.Int32` and that the number of times `OnEvict` is called will be 0 and occasionally 1, and  the fact that `OnEvict` is almost guaranteed to be called in a non-contentious situation anyway so it _really_ doesn't matter, I'm going to switch to `atomic.Int32`.